final-project/static/scripts.js

```diff
@@ -6,23 +6,59 @@
 // Enforce strict js code conventions
 "use strict";
 
-// Render product on screen
-function displayProduct() {
+// Render the product on screen.
+$(function retrieveJSON() {
```

Comment:
What is with this notation? Is it even valid?

Reply:
According to JQueries learning center, it is shorthand for "document ready"
`// Shorthand for $( document ).ready()`
`$(function() {`
`    console.log( "ready!" );`
`});`

I do not completely understand why I need it here, but the code does not work without it, and it does work with it. Maybe it is only a fluke it fixes the code. If you have any other insight, I am open to altering it.

Reply:
Should I be using JQuery? It seems like it might cause more headaches than it is worth. For example, it does not support promises. Rather, it supports Deferends which are similar, but different in a worse way. 

"Deferreds aren't Promise/A+ compliant, which makes them subtly different and less useful" - Google Developer docs

If I stuck with Javascript only, then would my code be more readable? It also seems like plain javascript syntax would be make it easier to apply other frameworks later on (e.g. React).

Please let me know what you think. I am not out in the field, so I am not sure what is a better standard to follow.

Reply:
You can still use JQuery without having to play with all its toys. It's a useful library that can get you far. I would still recommend programming declaratively and make your code composable and simple. You can sprinkle in some JQuery for when you need it in your implementation.